targets/TARGET_STM/TARGET_STM32F7/device/stm32f7xx_hal_def.h

```diff
@@ -130,19 +130,19 @@ static inline  void atomic_set_u32(volatile uint32_t *ptr, uint32_t mask)
 {
 	uint32_t newValue;
 	do {
-		newValue = (uint32_t)__LDREXW((volatile unsigned long *)ptr) | mask;
+		newValue = (uint32_t)__LDREXW(ptr) | mask;
```

Comment:
Are we eliminating warnings wit this commit  ? again nothing in the commit message.

How does this relate to cortex-a ?

Reply:
We are removing compilation failures. They change the signature in cmsis.

Reply:
@0xc0170 - This change will be for IAR8, same issue as seen with STM HAL https://github.com/ARMmbed/mbed-os/issues/5380